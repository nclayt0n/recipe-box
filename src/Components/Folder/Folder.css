* {
    box-sizing: border-box;
}

.folder {
    border: 1px solid green;
    padding: 20px;
    width: fit-content;
    margin: auto;
}

.folderRecipeList {
    padding: 10px;
    display: flex;
    width: fit-content;
}

.buttonBox {
    margin: auto;
    padding: 15px;
    width: fit-content;
}

h2 {
    text-align: center;
}