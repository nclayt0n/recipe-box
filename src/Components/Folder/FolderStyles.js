import backgroundImage from '../../images/purpleLpurpleleaves.jpg'

const folderStyles = {
    folder: {
        margin: '50px auto 50px',
        display: 'flex',
        flexDirection: 'column',
        width: '80%',

    },

    folderItems: {
        backgroundImage: `url(${backgroundImage})`,
        backgroundRepeat: 'repeat-y',
        backgroundSize: '100%',
        backgroundColor: 'var(--purple)',
        backgroundPosition: 'center',
        border: '1px solid #8a64a3',
    },

    folderRecipeList: {
        padding: '10px',
        color: 'white',
        textShadow: '1px 1px var(--darkPurple)',
        fontSize: '20px',
        textAlign: 'center',
        fontWeight: 'bolder'
    },

    folderRecipeListUl: {
        margin: 'auto',
        textAlign: 'center',

    },

    folderRecipeListLi: {
        width: '60%',
        listStyle: 'none',
        padding: '5px',
        margin: 'auto',
        textAlign: 'center',
        fontWeight: 'normal'
    },

    folderRecipeListLiA: {
        listStyle: 'none',
        textAlign: 'center',
        padding: '0px',
        color: 'white',
        textShadow: '.5px .5px #8a64a3',
        fontSize: '16px'
    },

    buttonBox: {
        display: 'flex',
        margin: 'auto',
        padding: '15px',
        width: 'fit-content'
    },
    buttonBoxMobile: {
        display: 'flex',
        margin: 'auto',
        padding: '15px',
        width: 'fit-content',
        flexDirection: 'column'
    },

    buttonBoxMobileButton: {
        fontSize: '14px',
        color: 'var(--purple)',
        border: '1px solid var(--purple)',
        fontWeight: 'bold'
    },

    buttonBoxMobileButtonA: {
        color: 'var(--purple)',
        fontWeight: 'bolder',
        fontSize: '14px'
    },
    tablet: {
        folder: {
            margin: '50px auto',
            display: 'flex',
            flexDirection: 'column',
            width: '60%',
        },
        folderRecipeListLi: {
            width: '60%',
            listStyle: 'none',
            padding: '5px',
            margin: 'auto',
            textAlign: 'center',
            fontWeight: 'normal'
        },
        folderRecipeListLiA: {
            listStyle: 'none',
            textAlign: 'center',
            padding: '0px',
            color: 'white',
            textShadow: '.5px .5px #8a64a3',
            fontSize: '16px'
        },

        folderItems: {
            backgroundImage: `url(${backgroundImage})`,
            backgroundRepeat: 'repeat-y',
            backgroundSize: '100%',
            backgroundColor: 'var(--purple)',
            backgroundPosition: 'center',
            border: '1px solid #8a64a3',
        },

        folderRecipeList: {
            padding: '10px',
            color: 'white',
            textShadow: '1px 1px var(--darkPurple)',
            fontSize: '18px',
            textAlign: 'center',
            fontWeight: 'bolder'
        },

        folderRecipeListUl: {
            margin: 'auto',
            textAlign: 'center'
        },

        buttonBox: {
            display: 'flex',
            margin: 'auto',
            padding: '15px',
            width: 'fit-content'
        },

        buttonBoxButton: {
            fontSize: '14px',
            margin: '5px',
            padding: '5px',
            color: 'var(--purple)',
            border: '1px solid var(--purple)',
            fontWeight: 'bolder'
        },

        buttonBoxButtonA: {
            color: 'var(--purple)',
            fontWeight: 'bolder'
        },
    },
    laptop: {
        folder: {
            margin: '50px auto',
            display: 'flex',
            flexDirection: 'column',
            width: '45%',
        },
        folderRecipeListLi: {
            width: '60%',
            listStyle: 'none',
            padding: '5px',
            margin: 'auto',
            textAlign: 'center',
            fontWeight: 'normal'
        },
        folderRecipeListLiA: {
            listStyle: 'none',
            textAlign: 'center',
            padding: '0px',
            color: 'white',
            textShadow: '.5px .5px #8a64a3',
            fontSize: '18px'
        },
        folderItems: {
            backgroundImage: `url(${backgroundImage})`,
            backgroundRepeat: 'repeat-y',
            backgroundSize: '100%',
            backgroundColor: 'var(--purple)',
            backgroundPosition: 'center',
            border: '1px solid #8a64a3',
        },

        folderRecipeList: {
            padding: '10px',
            color: 'white',
            textShadow: '1px 1px var(--darkPurple)',
            fontSize: '24px',
            textAlign: 'center',
            fontWeight: 'bolder'
        },

        folderRecipeListUl: {
            margin: 'auto',
            textAlign: 'center'
        },

        buttonBox: {
            display: 'flex',
            margin: 'auto',
            padding: '15px',
            width: 'fit-content'
        },

        buttonBoxButton: {
            fontSize: '16px',
            margin: '5px',
            padding: '5px',
            color: 'var(--purple)',
            border: '1px solid var(--purple)',
            fontWeight: 'bolder'
        },

        buttonBoxButtonA: {
            color: 'var(--purple)',
            fontWeight: 'bolder'
        },
    }
};
export default folderStyles;