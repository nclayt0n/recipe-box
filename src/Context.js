import React from 'react'

const Context = React.createContext({
    folders: [],
    recipes: [],
    user_id: '',
    addRecipe: () => {},
    addFolder: () => {},
    addFolders: () => {},
    addRecipes: () => {},
    updateRecipe: () => {},
    updateFolder: () => {},
    deleteRecipe: () => {},
    deleteFolder: () => {},

})
export default Context