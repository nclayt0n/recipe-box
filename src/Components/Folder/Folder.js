import React from 'react'
import {withRouter,Link} from 'react-router-dom'
import config from '../../config'
import Context from '../../Context'
import folderStyles from './FolderStyles'
import Nav from '../Nav/Nav'
import Header from '../Header/Header'
import TokenService from '../../services/token-service'
import ValidationError from '../../Validation/ValidationError'
import GetRecipeAndFolders from '../../Components/Network/GetRecipesAndFolders'
import MediaQuery from 'react-responsive'
const uuidv4 = require('uuid/v4');



class Folder extends React.Component{
    static contextType=Context;
    state={error:''}
    deleteFolder=(id,props)=>{
        const url=`${config.API_ENDPOINT}/folder/${id}`;
        const options={
            method:'DELETE',
            headers:{
          'content-type':'application/json',
          'Authorization': `Bearer ${TokenService.getAuthToken()}`,
        },
        body: JSON.stringify({'id':id})
    };
        fetch(url,options)
        .then(this.context.deleteFolder(id))
        .catch(error =>{
            this.setState({error})
        })
        props.history.push(`/home-page`)
    }
    findFolder=(id,folders)=>{
        if(folders.length===0){
            return null
        }else{
        const folder = folders.filter(f =>f.id === id );
            let folderName=(folder[0].name) 
            return folderName
        }
    }
    findRecipes=(folderId,recipes,style)=>{
        console.log(folderId)
        if(recipes.length===0){
            return null
        }else{
            let recipeList=recipes.filter(r=>r.folder_id===folderId);
            return recipeList.map(recipe=><>
            <li key={uuidv4()} style={style.folderRecipeListLi}><Link to={`/recipe/${recipe.id}`} style={style.folderRecipeListLiA}>{recipe.name}</Link></li></>)   
        }
    }
    findTabletRecipes=(folderId,recipes,style)=>{
        if(recipes.length===0){
            return null
        }else{
            let recipeList=recipes.filter(r=>r.folder_id===folderId);
            return recipeList.map(recipe=><>
            <li key={uuidv4()} style={style.tablet.folderRecipeListLi}><Link to={`/recipe/${recipe.id}`} style={style.tablet.folderRecipeListLiA}>{recipe.name}</Link></li></>)   
        }
    }
    render(){
        
        const style=folderStyles
        if(this.context.folders.length===0){
           return  <GetRecipeAndFolders/>
        }
        let id=parseInt(this.props.match.params.id)
        let folderName=this.findFolder(id,this.context.folders);
        let recipeList=this.findRecipes(id,this.context.recipes,style);
        console.log(recipeList)
        const tabletRecipeList=this.findTabletRecipes(id,this.context.recipes,style); 
        return(
            <>
                <Header/>
                <Nav/>
                <MediaQuery maxWidth={650}>
                    <div className='folder' style={style.folder}>
                        <div className='folderItems' style={style.folderItems}>
                            <ul className='folderRecipeList' key={uuidv4()} style={style.folderRecipeList}>
                                <h3 style={style.h}>{folderName}</h3>
                                    {recipeList}
                            </ul>
                            <div className='buttonBoxMobile' style={style.buttonBoxMobile}>
                                <button style={style.buttonBoxMobileButton}>
                                    <Link to={'/add-recipe'} style={style.buttonBoxMobileButtonA}>Add Recipe</Link>
                                </button><br/>
                                <button type='button' style={style.buttonBoxMobileButton}>
                                    <Link to={`/update-folder/${this.props.match.params.id}`} style={style.buttonBoxMobileButtonA}>Update Folder</Link>
                                </button><br/>
                                <button type='button' onClick={()=> this.deleteFolder(id,this.props)} style={style.buttonBoxMobileButton}> Delete Folder
                                </button><br/>
                            </div>
                            <ValidationError Namemessage={this.state.error}/><br/>
                        </div>
                    </div>
                </MediaQuery>
                <MediaQuery minWidth={651}>
                    <div className='folder' style={style.tablet.folder}>
                        <div className='folderItems' style={style.tablet.folderItems}>
                            <ul className='folderRecipeList' key={uuidv4()} style={style.tablet.folderRecipeList}>
                                <h3 style={style.tablet.h}>{folderName}</h3>
                                {tabletRecipeList}
                            </ul>
                            <div className='buttonBox' style={style.tablet.buttonBox}>
                                <button style={style.tablet.buttonBoxButton}>
                                    <Link to={'/add-recipe'} style={style.tablet.buttonBoxButtonA}>Add Recipe</Link>
                                </button><br/>
                                <button type='button' style={style.tablet.buttonBoxButton}>
                                    <Link to={`/update-folder/${parseInt(this.props.match.params.id)}`} style={style.tablet.buttonBoxButtonA}>Update Folder</Link>
                                </button><br/>
                                <button type='button' onClick={()=> this.deleteFolder(id,this.props)} style={style.tablet.buttonBoxButton}> Delete Folder
                                </button><br/>
                            </div> 
                            <ValidationError Namemessage={this.state.error}/><br/>
                        </div>
                    </div>
                </MediaQuery>
                </>
        )
    }
}
export default withRouter(Folder)