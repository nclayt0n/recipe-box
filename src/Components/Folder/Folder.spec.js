import React from 'react'
import ReactDOM from 'react-dom';
import Folder from './Folder'
import { shallow } from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import toJson from 'enzyme-to-json'
import * as Context from '../../Context'
 import ShallowRenderer from 'react-test-renderer/shallow';
 
 describe.skip('<Folder />', () => {
     const contextValues = { folders: [{
        "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fc3",
        "name": "Spangley",
        "user_id": 1
     },{
        "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fc3",
        "name": "Spangley",
        "user_id": 1
     }]}
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><Folder/></BrowserRouter>,div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('snapshot', () => {
const wrapper=shallow(<Folder contextValues={contextValues}/>)
expect(toJson(wrapper))
.toMatchSnapshot()
     console.log(Context)
});
     })

