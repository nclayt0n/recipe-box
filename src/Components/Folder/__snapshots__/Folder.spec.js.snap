// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Folder /> snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
